code/game/objects/items/weapons/boomerang.dm

```diff
@@ -1,25 +1,278 @@
+
+//////////////////////////////BOOMERANG ITEM////////////////////////////////
+
 /obj/item/weapon/boomerang
 	name = "boomerang"
 	desc = "A heavy, curved piece of wood used by Space Australians for hunting, sport, entertainment, cooking, religious rituals and warfare. When thrown, it will either deal a devastating blow to somebody's head, or return back to the thrower." //also used for shitposting
+	icon = 'icons/obj/boomerang.dmi'
 	icon_state = "boomerang"
+	item_state = null
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/boomerang.dmi', "right_hand" = 'icons/mob/in-hand/right/boomerang.dmi')
 
 	w_class = W_CLASS_MEDIUM
-
-	throwforce = 16
+	force = 7
+	throwforce = 10
 	throw_range = 7
 	throw_speed = 5
 
-	force = 7
-
 	starting_materials = list(MAT_WOOD = CC_PER_SHEET_WOOD * 6)
 	w_type = RECYK_WOOD
 	flammable = TRUE
 
-/obj/item/weapon/boomerang/Destroy()
+	var/mob/living/carbon/originator = null
+	var/thrown = FALSE
+	var/throw_mult = 1
+	var/stun = 0
+	var/weaken = 2
+	var/sound_throw = 'sound/weapons/boomerang_start.ogg'
+	var/sound_loop = 'sound/weapons/boomerang_loop.ogg'
+	var/sound_bump = 'sound/weapons/kick.ogg'
+	var/last_sound_loop = 0
+
+/obj/item/weapon/boomerang/proc/on_step(var/obj/O)
+	if (world.time > (last_sound_loop+10))
+		last_sound_loop = world.time
+		playsound(loc,sound_loop, 35, 0)
+
+/obj/item/weapon/boomerang/proc/return_check()//lets you add conditions for the boomerang to come back
+	return TRUE
+
+/obj/item/weapon/boomerang/proc/apply_status_effects(var/mob/living/carbon/C, var/minimal_effect = 0)
+	if (ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(istype(H.head, /obj/item/clothing/head/helmet))
```

Comment:
why not an armor check?

Reply:
I changed it to protection from stuns being granted should you be wearing any hat that provides some armor to melee or bullet damage, so you can use metal buckets as well. I prefer that to an RNG armor check.